Update EIP-7997: remove specification of nonce (#11846)
ethereum/EIPs: EIPS/eip-7997.md

* `FACTORY_ADDRESS` = `0x4e59b44847b379578588920cA78FbF26c0B4956C`

> The chain's state MUST include `FACTORY_ADDRESS` with this runtime code:

Reviewer: ```suggestion
The chain's state MUST include `FACTORY_ADDRESS` with nonzero nonce and this runtime code:
```

Reviewer: Added this in https://github.com/ethereum/EIPs/pull/11852.

```
0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3
```

> A chain may satisfy this requirement by any means that results in the account above: deploying the factory with an ordinary transaction (for example, its keyless creation transaction) where the chain's gas parameters permit it, including the account in the genesis state, or inserting it via an irregular state transition. Regardless of the method used, the resulting `FACTORY_ADDRESS` account MUST have the runtime code specified above. If deployed using an irregular state transition, the nonce SHOULD be set to 1.

Reviewer 2: Since this already caused a chain split in a devnet, perhaps it's worth it to make it explicit here:
```suggestion
A chain may satisfy this requirement by any means that results in the account above: deploying the factory with an ordinary transaction (for example, its keyless creation transaction) where the chain's gas parameters permit it, including the account in the genesis state, or inserting it via an irregular state transition. Regardless of the method used, the resulting `FACTORY_ADDRESS` account MUST have the runtime code specified above. If deployed using an irregular state transition, the nonce SHOULD be set to 1. Clients MUST NOT reset the nonce to 1 if it is already higher.
```

Reviewer: My suggestion to make it fully specified:

```suggestion
A chain may satisfy this requirement by any means that results in the account above: deploying the factory with an ordinary transaction (for example, its keyless creation transaction) where the chain's gas parameters permit it, including the account in the genesis state, or inserting it via an irregular state transition. Regardless of the method used, the resulting `FACTORY_ADDRESS` account MUST have the runtime code specified above. When deployed using an irregular state transition, if the account nonce is 0, it MUST be set to 1, and if the account nonce is already nonzero, it MUST NOT be modified.
```
